### PR TITLE
Corrections / New Types

### PR DESCRIPTION
**Pull request by thomasduenser**

### CODE REVIEW

File: Entity/Dynamic.php [hunk 1]
```diff
@@ -170,6 +170,10 @@ public function getData()
      */
     public function __set($name, $value)
     {
+        if ($value instanceof \DateTime) {
```
Comment (alexander-schranz) [inline Entity/Dynamic.php]: is this needed when `property_exists`?

Comment (thomasduenser) [inline Entity/Dynamic.php]: @alexander-schranz Yes, because `date` is defined as `string` in the XML. We could change that, but then we have two different cases -> if property_exists, it's a DateTime object, if not, it's a string...

**Comment by dominikmatt** [comment]
@thomasduenser ACK

File: Entity/Dynamic.php (hunk 1 above)
Comment (alexander-schranz) [inline Entity/Dynamic.php]: save it here as datetime (update the orm file)  and fix the __get function when its from type `date` to convert it back to a dateTime.

So add something like this to #L208

```
if (strpos($name, 'date') === 0) {
   return new \DateTime($array[$name]);
}
```

Test some edge cases.

File: Entity/Dynamic.php [hunk 2]
```diff
@@ -170,6 +176,10 @@ public function getData()
...
+            $value = $value->format('Y-m-d H:i:s');
+        }
+
         if (property_exists($this, $name)) {
             if (is_array($value)) {
                 $value = json_encode($value);
```
Comment (alexander-schranz) [inline Entity/Dynamic.php:231]: in `__get` #L203 the json_decode($value, true) is missing when type `multipleChoice`.

**Comment by dominikmatt** [comment]
@thomasduenser ACK
